packages/gil/lib/structures/Task.ts

```diff
@@ -0,0 +1,16 @@
+import type { BotClient } from "../BotClient";
+
+export abstract class Task {
+    /** The amount of time this task should take to run. Defaults to 1 hour(3,600,000 ms) */
+    millisecondsInterval = 60 * 60 * 100;
+    /** Whether or not this task should run immediately on startup. Default to false. */
+    runOnStartup = false;
+    /** Whether this task requires the bot to be fully ready before running. Default to false. */
+    requireFullyReady = false;
```

Comment:
I'm being a little picky here, but I feel like `requireFullyReady` is too verbose, can be simplified to `requireReady`